drop-in/src/main/java/com/adyen/checkout/dropin/ComponentParsingProvider.kt

```diff
@@ -444,51 +445,45 @@ internal fun getActionProviderFor(
     return allActionProviders.firstOrNull { it.canHandleAction(action) }
 }
 
-/**
- * Provides a [BaseActionComponent] from an [ActionComponentProvider].
- *
- * @param activity The Action which the ActionComponent lifecycle will be bound to.
- * @param provider The component provider.
- * @throws CheckoutException In case a component cannot be created.
- */
 @Suppress("ComplexMethod", "LongMethod")
-internal fun getActionComponentFor(
-    activity: FragmentActivity,
+internal fun <T> getActionComponentFor(
```

Comment:
I guess this has nothing to do with StatusRepository refactor, so it's better to be done in a separate PR.